packages/swarm-components/src/Dropdown.jsx

```diff
@@ -0,0 +1,462 @@
+/* eslint-disable */
+import React, {
+	createContext,
+	Children,
+	useContext,
+	useState,
+	useEffect,
+	useRef,
+} from 'react';
+import Portal from '@reach/portal';
+import Rect, { useRect } from '@reach/rect';
+import WindowSize from '@reach/window-size';
+import { node, func, object, string, number, oneOfType, any } from 'prop-types';
+import { wrapEvent, checkStyles, assignRef } from '@reach/utils';
+import { ForwardedButton } from './Button';
+
+// TODO: add the mousedown/drag/mouseup to select of native menus, will
+// also help w/ remove the menu button tooltip hide-flash.
+
+// TODO: add type-to-highlight like native menus
+
+let MenuContext = createContext();
+
+let checkIfAppManagedFocus = ({ refs, state }) => {
+	if (!state.isOpen && !!refs.menu) {
```

Comment:
why `!!` here? I think you can avoid it, but in general I would use `Boolean(...)` instead of `!!` just because the intent is a bit more explicit and it's less prone to typos.